// weatherly/js/model/TodaysWeather.js
'use strict';

var TodaysWeather = Backbone.Model.extend({
    initialize: function () {
        this.attributes.temperature = this.convertTemperature(this.get('country'), this.get('temperature'));
        this.attributes.apparentTemperature = this.convertTemperature(this.get('country'), this.get('apparentTemperature'));

        this.on('change:temperature', this.temperatureChanged, this);
        this.on('change:apparentTemperature', this.apparentTemperatureChanged, this);
    },
    temperatureChanged: function () {
        this.attributes.temperature = this.convertTemperature(this.get('country'), this.get('temperature'));
    },
    apparentTemperatureChanged: function () {
        this.attributes.apparentTemperature = this.convertTemperature(this.get('country'), this.get('apparentTemperature'));
    },
    convertTemperature: function (country, temperature) {
        if(this.isCelsiusCountry(country)) {
            return this.convertToCelsius(temperature);
        }

        return temperature;
    },
    isCelsiusCountry: function (country) {
        return ['BS', 'BZ', 'KY', 'PW', 'US', 'AS', 'VI'].indexOf(country) === -1;
    },
    convertToCelsius: function (temperature) {
        return (temperature - 32) * (5/9);
    }
});

module.exports = TodaysWeather;
